warehouse/query-core/src/test/java/datawave/query/testframework/AbstractFunctionalQuery.java

```diff
@@ -398,11 +403,15 @@ protected void runTestQuery(Collection<String> expected, String queryStr, Date s
         q.setId(UUID.randomUUID());
         q.setPagesize(Integer.MAX_VALUE);
         q.setQueryAuthorizations(auths.toString());
-        
-        GenericQueryConfiguration config = this.logic.initialize(connector, q, this.authSet);
-        this.logic.setupQuery(config);
-        if (log.isDebugEnabled()) {
-            log.debug("Plan: " + config.getQueryString());
+        if (useRunningQuery) {
+            QueryMetricFactory queryMetricFactory = (metricFactory == null) ? new QueryMetricFactoryImpl() : metricFactory;
+            new RunningQuery(connector, AccumuloConnectionFactory.Priority.NORMAL, this.logic, q, "", principal, queryMetricFactory);
+        } else {
+            GenericQueryConfiguration config = this.logic.initialize(connector, q, this.authSet);
+            this.logic.setupQuery(config);
+            if (log.isDebugEnabled()) {
+                log.debug("Plan: " + config.getQueryString());
+            }
```

Comment:
I need to dig into this a little deeper, but at first glance I don't love the idea of modifying AbstractFunctionalQuery to support testing, assuming that is what's going on here.

Reply:
Despite its name, AbstractFunctionalQuery exists solely for testing.

Reply:
ah, got it.  my mistake.  the name definitely threw me.  proceed as-is then.  thanks!